text/0000-remove-stacks-mixins.md

```diff
@@ -0,0 +1,139 @@
+
+App image builds fail if the build image and selected run image have mismatched metadata. We may introduce flags or additional labels to skip this validation (e.g., for cross-compilation or minimal runtime base images). An image without a specified Distribution is compatible with images specifying any Distribution. An image specifying a Distribution without a Version is compatible with images specifying any Versions of that Distribution.
+
+When an app image is rebased, `pack rebase` will fail if the new run image and previous run image have mismatched metadata. This check may be skipped for Distribution and Version by passing a new `--force` flag to `pack rebase`.
+
+#### Example: buildpack.toml `targets` table
+
+```toml
+[[targets]]
+os = "linux"
+arch = "x86_64"
+[[targets.distros]]
+name = "ubuntu"
+[[targets.distros.versions]]
+version = "18.04"
+packages = ["pkg:deb/ubuntu/curl"]
+[[targets.distros.versions]]
+version = "20.04"
+packages = ["pkg:deb/ubuntu/curl2"]
+
+[[targets]]
+os = "linux"
+arch = "x86_64"
+[[targets.distros]]
+name = "ubuntu"
+packages = ["pkg:deb/ubuntu/curl"]
+[[targets.distros.versions]]
+version = "14.04"
+[[targets.distros.versions]]
+version = "16.04"
+```
+
+## Mixins
+
+The mixins label on each base image is replaced by a layer in each base image containing a single file consisting of a [CycloneDX](https://cyclonedx.org)-formatted list of packages. Each package entry has a [PURL](https://github.com/package-url/purl-spec)-formatted ID that uniquely identifies the package.
+
+### Example: CycloneDX package list
+
+```json
+{
+  "bomFormat": "CycloneDX",
+  "specVersion": "1.3",
+  "version": 1,
+  "components": [
+    {
+      "type": "library",
+      "name": "curl",
+      "version": "1.2.3",
+      "purl": "pkg:deb/ubuntu/curl@1.2.3"
+    },
+    {
+      "type": "library", 
+      "name": "libcurl",
+      "version": "4.5.6",
+      "purl": "pkg:deb/ubuntu/libcurl@4.5.6"
+    }
+  ]
+}
+```
+
+### Validations
+
+Buildpack base image metadata and packages specified in `buildpack.toml`'s `targets` list are validated against the runtime and build-time base images.
+
+Runtime and build-time base image packages are no longer validated against each other.
+
+When an app image is rebased, `pack rebase` will fail if packages are removed from the new runtime base image. This check may be skipped by passing a new `--force` flag to `pack rebase`.
+
+## Runtime Base Image Requirements
+
+Buildpacks may specify a list of package names (as PURL URLs without versions or qualifiers) in a `packages` table in the build plan. The build will fail if these packages are not present in the runtime base image.
+
```

Comment:
I think we need to address the migration path in this RFC

Reply:
Addressed, let me know if you think I covered the relevant migration scenarios

Reply:
Thanks for adding this! Would we also continue to allow `[[stacks]]` in `buildpack.toml`, and ignore it at some point?

Reply:
Definitely, I can add that as well

Reply:
> These labels will be deprecated (but allowed) for the forseeable future.

@jkutner This makes me think `[[stacks]]` and those labels would go away at the same time.